.github/workflows/wheel.yaml

```diff
@@ -1,39 +1,62 @@
 name: Create and publish Python package
 
 on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
   push:
+    tags:
+      - "v*"
+    branches:
+      - "main"
+  workflow_dispatch:
 
 jobs:
+  version:
+    uses: ./.github/workflows/versioning.yaml
+
   build:
-    runs-on:
-      - ubuntu-latest
+    runs-on: ubuntu-latest
+    needs: version
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
-        with:
-          fetch-tags: true
       - uses: actions/setup-python@v5
         with:
           python-version: '3.12'
+          cache: pip
+      - name: Install build tool
+        uses: pypa/hatch@install
       - name: Build wheel
         working-directory: python
-        run: pip wheel --no-deps -w dist .
+        env:
+          SETUPTOOLS_SCM_PRETEND_VERSION: ${{ needs.version.outputs.pep440 }}
+        run: hatch build -t wheel
       - uses: actions/upload-artifact@v4
         with:
-          path: python/dist/*.whl
-  # TODO: Uncomment me when GB PyPI account is setup
-  # publish:
-  #   name: Upload release to PyPI
-  #   runs-on:
-  #     - gb-arc-staging
-  #   environment:
-  #     name: pypi
-  #     url: https://pypi.org/p/greenbids-tailor
-  #   permissions:
-  #     id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
-  #   steps:
-  #     - uses: actions/download-artifact@v4
-  #       with:
-  #         path: dist/
-  #     - name: Publish package distributions to PyPI
-  #       uses: pypa/gh-action-pypi-publish@release/v1
+          name: wheel
+          path: python/dist/
+
+  publish:
+    name: Upload release to PyPI
+    needs: build
+    if: github.event_type == 'push'
```

Comment:
suggestion: when we trigger the publish of wheel package can add the restriction the action only for push into main branches like  `github.ref == 'refs/heads/main`